Apps/Database/src/Delegates/DBUserPreferenceDelegate.cs

```diff
@@ -72,7 +72,7 @@ public DBResult<IEnumerable<UserPreference>> SaveUserPreferences(string hdid, IE
 
             foreach (UserPreference preference in preferences.Payload)
             {
-                if (!newPreferences.Any(p => p.Preference == preference.Preference))
```

Comment:
This fix to avoid overwriting an existing user preference.

Reply:
Why are we reading the preference from the DB in order to update the preference.  We shouldn't have to do this as we should have read the preference in the first place to determine our next steps.  I also don't get why we have a looped remove... The logic to either add, update, or remove should be at the service level.

Reply:
@sslaws Good point, should we create a bug to address this existing design/code or should I address this one as a part of this PR?